lib/plugins/keyGeneratorFactory.js

```diff
@@ -0,0 +1,206 @@
+'use strict';
+// TODO: should be instance! cannot use Nocca instance references now.
+
+var $q = require('q');
+var $extend = require('extend');
+
+module.exports = {
+    interface: 'keyGenerator',
+    id: 'keyGeneratorFactory',
+    name: 'Key Generator Factory',
+    constructor: KeyGeneratorFactory
+};
+
+function KeyGeneratorFactory (Nocca) {
+
+    var self = this;
+    
+    self.generateKey = generateKey;
+
+    self.incrementalKeyGenerator = incrementalKeyGenerator;
```

Comment:
It could be split up, but then it would lose the 'combine' feature I guess? The config object (Nocca global config or per-request config) is accessible from the plugin, so plugin specific configuration can be added (which currently is also used for this plugin). Could be improved to use i.e. an array of the methods to combine instead of having to reference a function directly.